File: js/01-hashing/021-all-unique.js
function allUnique(l) {
    let all = {};

    for (item in l) {
        if (all[l[item]]) {
            return false;
        } else {
            all[l[item]] = 1;
        }
    };

    return true;
};

// test cases
console.log(allUnique(["q", "r", "s", "a"])) // -> True
console.log(allUnique(["q", "r", "s", "a", "r", "z"])) // -> False
console.log(allUnique(["red", "blue", "yellow", "green", "orange"])) // -> True
console.log(allUnique(["cat", "cat", "dog"])) // -> False
console.log(allUnique(["a", "u", "t", "u", "m", "n"])) // -> False
console.log(allUnique("spring".split())) // -> True

File: py/01-hashing/021-all-unique.py
def all_unique(l):
    all_items = {}

    for item in l:
        if item in all_items:
            return False
        else:
            all_items[item] = 1

    return True


# test cases
print(all_unique(["q", "r", "s", "a"])) # -> True
print(all_unique(["q", "r", "s", "a", "r", "z"])) # -> False
print(all_unique(["red", "blue", "yellow", "green", "orange"])) # -> True
print(all_unique(["cat", "cat", "dog"])) # -> False
print(all_unique(["a", "u", "t", "u", "m", "n"])) # -> False
print(all_unique("spring".split())) # -> True